apps/api_web/test/api_web/controllers/schedule_controller_test.exs

```diff
@@ -156,11 +156,78 @@ defmodule ApiWeb.SchedulerControllerTest do
       assert index_data(conn, %{}) == {:error, :filter_required}
     end
 
+    test "returns an error if only the route_type filter is provided", %{conn: conn} do
+      assert ApiWeb.ScheduleController.index_data(conn, %{"route_type" => "0,1"}) ==
+               {:error, :only_route_type}
+    end
+
     test "can filter by stop", %{conn: conn} do
       assert index_data(conn, %{"stop" => "stop"}) == [@schedule]
       assert index_data(conn, %{"stop" => "not stop"}) == []
     end
 
+    test "can filter by route_type", %{conn: conn} do
+      schedule1 = %Model.Schedule{
+        route_id: "1",
+        trip_id: "trip",
+        stop_id: "stop",
+        direction_id: 1,
+        arrival_time: 45_000,
+        departure_time: 45_100,
+        drop_off_type: 0,
+        pickup_type: 0,
+        timepoint?: false,
+        service_id: "service",
+        stop_sequence: 2,
+        position: :first
+      }
+
+      schedule2 = %Model.Schedule{
+        route_id: "2",
+        trip_id: "trip",
+        stop_id: "stop",
+        direction_id: 1,
+        arrival_time: 45_000,
+        departure_time: 45_100,
+        drop_off_type: 0,
+        pickup_type: 0,
+        timepoint?: false,
+        service_id: "service",
+        stop_sequence: 2,
+        position: :first
+      }
+
+      route1 = %Model.Route{
+        id: "1",
+        type: 1,
+        sort_order: 1,
+        description: "First Route",
+        short_name: "First",
+        long_name: "First"
+      }
+
+      route2 = %Model.Route{
+        id: "2",
+        type: 2,
+        sort_order: 1,
+        description: "Second Route",
+        short_name: "Second",
+        long_name: "Second"
+      }
+
+      State.Route.new_state([route1, route2])
+      State.Schedule.new_state([schedule1, schedule2])
+
+      for {params, expected} <- [
+            {%{"stop" => "stop"}, [schedule1, schedule2]},
+            {%{"stop" => "stop", "route_type" => "2"}, [schedule2]},
+            {%{"stop" => "stop", "route_type" => "1,2"}, [schedule1, schedule2]}
+          ] do
+        conn = get(conn, "/schedules", params)
+        assert conn.assigns.data == expected
+      end
```

Comment:
Any strong opinions on using a loop like this in tests vs. writing out each assertion? I favor the latter since if a test fails it can be hard to tell which iteration of a loop it failed on. For example, here if the failure message tells you the `expected` was `[schedule1, schedule2]`, you don't know if it was on the first or the third iteration. The equivalent "unrolled" version doesn't seem _too_ verbose to me:

```elixir
conn = get(conn, "/schedules", %{"stop" => "stop"})
assert conn.assigns.data == [schedule1, schedule2]

conn = get(conn, "/schedules", %{"stop" => "stop", "route_type" => "2"})
assert conn.assigns.data == [schedule2]

conn = get(conn, "/schedules", %{"stop" => "stop", "route_type" => "1,2"})
assert conn.assigns.data == [schedule1, schedule2]
```

Reply:
no strong opinions. i think the above looks fine too, so i'll do that since it produces a more useful message.